src/validators.js

```diff
@@ -1,20 +1,44 @@
 
 const assert = require('assert');
 
+
 function validateApiKey(value) {
-    const typeError = new TypeError('ImgixAPI.settings.apiKey must be passed a string');
+    const invalidApiKeyError = new TypeError('ImgixAPI.settings.apiKey must be passed a string');
     const legalKey = /[0-9a-f]{64}/;
     const legalKeyError = new TypeError(`${value} does not match a legal apiKey structure`);
 
-    assert(typeof value === 'string', typeError);
+    assert(typeof value === 'string', invalidApiKeyError);
     assert(legalKey.exec(value), legalKeyError);
 };
 
 function validateOpts(options) {
     validateApiKey(options.apiKey);
 };
 
+function validateBody(body) {
+    const invalidBodyError = new TypeError('The request body must be of type JSON or Buffer');
+
+    const isValid = body && (isJSON(body) || isBuffer(body));
+    assert(isValid, invalidBodyError);
+};
+
+function isBuffer(body) {
+    return Buffer.isBuffer(body);
```

Comment:
The `Buffer` object is not available in browser environments, so

- Using this here means this library is Node-only, which means
- We won't be able to pull this library into imgix-core-js without some refactoring.

I don't think this should block the launch of this experimental project, but it does represent down-the-line work that we should be aware of.

Reply:
Oh, this also obviates the need to do fetch-wrapper shenanigans, since we won't be supporting browser envs anyway.

Reply:
Agreed, it's part of a growing list of items that need to be addressed when we to start supporting this in the browser.